src/odemis/driver/ueye.py

```diff
@@ -760,10 +766,15 @@ def _set_static_settings(self):
         # Note: for now the default pixel clock seems fine. Moreover, reducing
         # it reduces the minimum exposure time (but we don't care), and the
         # maximum exposure time is constant.
+        self.BlackLevel()
 
-        # Auto black level = different black level per pixel (= more range?)
-        blmode = c_uint32(AUTO_BLACKLEVEL_ON)
-        self._dll.is_Blacklevel(self._hcam, BLACKLEVEL_CMD_SET_MODE, byref(blmode), sizeof(blmode))
+    def BlackLevel(self):
+        try:
+            # Auto black level = different black level per pixel (= more range?)
+            blmode = c_uint32(AUTO_BLACKLEVEL_ON)
+            self._dll.is_Blacklevel(self._hcam, BLACKLEVEL_CMD_SET_MODE, byref(blmode), sizeof(blmode))
+        except Exception as e:
```

Comment:
What are you trying to hide here? IIRC, that's in case the camera doesn't support it? If so, it should _only_ catch `UEyeError`, and preferably checking for a specific `errno`. You might also want to instead check the features of the camera ahead of it (and skip if auto black level is not supported). Or, alternatively, move this exception handler to the caller.